classes/controllers/deadline_extensions_controller.php

```diff
@@ -39,13 +39,14 @@
  * @property deadline_extension_form form
  * @package mod_coursework\controllers
  */
+#[\AllowDynamicProperties]
 class deadline_extensions_controller extends controller_base {
 
     protected function show_deadline_extension() {
         global $USER, $PAGE;
 
         $ability = new ability(user::find($USER), $this->coursework);
-        $ability->require_can('show', $this->deadline_extension);
+        $ability->require_can('show', $this->deadlineextension);
```

Comment:
Class property should be `deadlineextension`
https://github.com/ucl-isd/moodle-mod_coursework/pull/32/files#diff-85cf92872e60603c1d6eaf74e5d34ff7c507a76f90fd94191ada7ccf21598e09L37